Replace Less z-index (`@zi-*`) variables with CSS variables

dancormier opened the pull request:
**Note**: This is a breaking change for in many spots in Core.

dancormier committed:
Replace Less `@zi-*` vars with CSS vars

aaronshekey commented on lib/css/exports/_stacks-constants-helpers.less:
```diff
@@ -21,18 +21,20 @@
+    --zi-navigation:         5000;   //  Navigation Bars
+    --zi-navigation-fixed:   5050;   //  Fixed navigation bars
+    --zi-modals:             9000;   //  Modals
+    --zi-modals-background:  calc(var(--zi-modals) - 50);
```
Maybe we just hard code this. It's not like `zi-modals` varies.

aaronshekey commented on lib/css/exports/_stacks-constants-helpers.less:
```diff
@@ -21,18 +21,20 @@
-@zi-navigation-fixed:   5050;   //  Fixed navigation bars
-@zi-modals:             9000;   //  Modals
-@zi-modals-background:  @zi-modals - 50;
+body {
```
Some discussion might need to be had about where we actually put some of these. I don't imagine a theme or a user modifying these specific variables, so they could be tied to `:root`? 🤔 

This works for now, of course.

dancormier committed:
Merge branch 'develop' into less-to-css-vars-zi

dancormier committed:
Hardcode zi-modals-background value

Addresses https://github.com/StackExchange/Stacks/pull/895/#discussion_r839009612
